File: software/examples/java/ExampleCallback.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletDualButtonV2;

public class ExampleCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Dual Button Bricklet 2.0
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletDualButtonV2 db = new BrickletDualButtonV2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add state changed listener
		db.addStateChangedListener(new BrickletDualButtonV2.StateChangedListener() {
			public void stateChanged(int buttonL, int buttonR, int ledL, int ledR) {
				if(buttonL == BrickletDualButtonV2.BUTTON_STATE_PRESSED) {
					System.out.println("Left button pressed");
				} else {
					System.out.println("Left button released");
				}

				if(buttonR == BrickletDualButtonV2.BUTTON_STATE_PRESSED) {
					System.out.println("Right button pressed");
				} else {
					System.out.println("Right button released");
				}

				System.out.println("");
			}
		});

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/go/example_callback.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/dual_button_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Dual Button Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	db, _ := dual_button_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	db.RegisterStateChangedCallback(func(buttonL dual_button_v2_bricklet.ButtonState, buttonR dual_button_v2_bricklet.ButtonState, ledL dual_button_v2_bricklet.LEDState, ledR dual_button_v2_bricklet.LEDState) {

		if buttonL == dual_button_v2_bricklet.ButtonStatePressed {
			fmt.Println("Left Button: Pressed")
		} else if buttonL == dual_button_v2_bricklet.ButtonStateReleased {
			fmt.Println("Left Button: Released")
		}

		if buttonR == dual_button_v2_bricklet.ButtonStatePressed {
			fmt.Println("Right Button: Pressed")
		} else if buttonR == dual_button_v2_bricklet.ButtonStateReleased {
			fmt.Println("Right Button: Released")
		}

		fmt.Println()
	})

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
